File: lifetrackhub-be/src/main/java/com/lifetrackhub/entity/File.java
package com.lifetrackhub.entity;

import com.lifetrackhub.constant.enumeration.FileType;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Data
@ToString
@Entity
@Table
public class File {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private Long userId;

    @NotNull
    @Enumerated(EnumType.STRING)
    private FileType fileType;

    @NotNull
    private String originalFileName;

    @NotNull
    private String filePath;

    @NotNull
    private String previewUrl;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private Instant createdDate;

    @UpdateTimestamp
    @Column(nullable = false)
    private Instant lastModifiedDate;
}

File: lifetrackhub-fe-v2/src/types/file.ts
export type FileType = 'IMG' | 'PDF';

export interface FileDto {
  userId: number;
  fileType: FileType;
  originalFileName: string;
  filePath: string;
  previewUrl: string;
  createdDate: string;
  lastModifiedDate: string;
}

export interface PageDto<T> {
  content: T[];
  pageNumber: number;
  hasNext: boolean;
  hasPrevious: boolean;
  totalPages: number;
}
